pyscraper/new_hansard.py

```diff
@@ -0,0 +1,1442 @@
+                    fragment=member_tag.text
+                )
+
+        return None
+
+    def parse_date(self, date):
+        text = self.get_single_line_text_from_element(date)
+        time_parts = re.match('\s*the\s+house (?:being |having )?met at?\s+(.*?)$(?i)', text)
+        if time_parts:
+            time = time_parts.group(1)
+            time = re.sub('</?i>',' ', time)
+            time = re.sub('\s+',' ', time)
+            if re.match("half-past Nine(?i)", time):
+                    newtime = '09:30:00'
+            elif re.match("a quarter to Ten o(?i)", time):
+                    newtime = '09:45:00'
+            elif re.match("Ten o'clock(?i)", time):
+                    newtime = '10:00:00'
+            elif re.match("half-past Ten(?i)", time):
+                    newtime = '10:30:00'
+            elif re.match("Eleven o&#039;clock(?i)", time):
+                    newtime = '11:00:00'
+            elif re.match("twenty-five minutes past\s*Eleven(?i)", time):
+                    newtime = '11:25:00'
+            elif re.match("twenty-six minutes past\s*Eleven(?i)", time):
+                    newtime = '11:26:00'
+            elif re.match("twenty-nine minutes past\s*Eleven(?i)", time):
+                    newtime = '11:29:00'
+            elif re.match("half-past Eleven(?i)", time):
+                    newtime = '11:30:00'
+            elif re.match("Twelve noon(?i)", time):
+                    newtime = '12:00:00'
+            elif re.match("half-past One(?i)", time):
+                    newtime = '13:30:00'
+            elif re.match("half-past Two(?i)", time):
+                    newtime = '14:30:00'
+            elif re.match("twenty minutes to Three(?i)", time):
+                    newtime = '14:40:00'
+            elif re.match("10 minutes past Three(?i)", time):
+                    newtime = '15:10:00'
+            elif re.match("Six o'clock(?i)", time):
+                    newtime = '18:00:00'
+
+            self.time = newtime
+        elif re.match('.*was asked.*', text):
+            self.parse_para_with_member(date, None)
+
+    def parse_oral_heading(self, heading):
+        self.output_heading = True
+        tag = etree.Element('oral-heading')
+        tag.set('id', self.get_speech_id())
+        tag.set('nospeaker', 'true')
+        tag.set('colnum', self.current_col)
+        tag.set('time', self.current_time)
+        tag.set(
+            'url',
+            'http://www.publications.parliament.uk{0}'.format(
+                heading.get('url')
+            )
+        )
+        tag.text = heading.text
+        self.root.append(tag)
+
+    def parse_major(self, heading, **kwargs):
+        self.check_for_pi(heading)
+        text = self.get_text_from_element(heading)
+        if text.strip() == 'Prayers':
+            return
+        self.clear_current_speech()
+        tag = etree.Element('major-heading')
+        if self.uc_titles:
+            tag.text = text.upper()
```

Comment:
This is presumably as the TWFY loading in needs this to detect properly? Do we want to not lose information here (e.g. might lose acronyms) and update the importer now it can hopefully rely on its source being a bit better?